packages/techdocs-common/src/stages/publish/local.ts

```diff
@@ -0,0 +1,144 @@
+/*
+ * Copyright 2020 Spotify AB
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import fetch from 'cross-fetch';
+import express from 'express';
+import fs from 'fs-extra';
+import { Logger } from 'winston';
+import { Entity, EntityName } from '@backstage/catalog-model';
+import {
+  resolvePackagePath,
+  PluginEndpointDiscovery,
+} from '@backstage/backend-common';
+import { Config } from '@backstage/config';
+import { PublisherBase, PublishRequest } from './types';
+
+/* `remoteUrl` is the URL which serves files from the local publisher's static directory. */
+export type LocalPublishReturn = Promise<{ remoteUrl: string }>;
+
+const staticDocsDir = resolvePackagePath(
+  '@backstage/plugin-techdocs-backend',
+  'static/docs',
+);
+
+/**
+ * Local publisher which uses the local filesystem to store the generated static files. It uses a directory
+ * called "static" at the root of techdocs-backend plugin.
+ */
+export class LocalPublish implements PublisherBase {
+  private readonly config: Config;
+  private readonly logger: Logger;
+  private readonly discovery: PluginEndpointDiscovery;
+
+  constructor(
+    config: Config,
+    logger: Logger,
+    discovery: PluginEndpointDiscovery,
+  ) {
+    this.config = config;
+    this.logger = logger;
+    this.discovery = discovery;
+  }
+
+  publish({ entity, directory }: PublishRequest): LocalPublishReturn {
```

Comment:
Seems like the response here differs from the response for the GCS equivalent method.  Should they conform to the same interface?

Reply:
Yep, good catch. :)